### PR TITLE
Add logic to handle missing award amount from LH response

### PR DESCRIPTION
**Pull request by kjduensing**
If a Veteran does not have a monthly award amount the Lighthouse letters generator API excludes the key from the response. This requires the va.gov letters generator service to handle that case.

Closes department-of-veterans-affairs/va.gov-team#60321

### CODE REVIEW

File: lib/lighthouse/letters_generator/service.rb [hunk 1]
```diff
@@ -53,31 +63,23 @@ def transform_military_services(services_info)
...
-          specialMonthlyCompensation: :hasSpecialMonthlyCompensation
-        }
-
         symbolized_info = info.deep_transform_keys(&:to_sym)
 
         transformed_info = symbolized_info.reduce({}) do |acc, (k, v)|
-          if transform_targets.key? k
-            acc.merge({ transform_targets[k] => v })
+          if BENFICIARY_KEY_TRANFORMS.key? k
+            acc.merge({ BENFICIARY_KEY_TRANFORMS[k] => v })
```
Comment (ryan-mcneil) [inline lib/lighthouse/letters_generator/service.rb:70]: spelling, if I'm being picky :)

**Review by ryan-mcneil — APPROVED** [review]
👌

File: lib/lighthouse/letters_generator/service.rb (hunk 1 above)
Comment (kjduensing) [reply lib/lighthouse/letters_generator/service.rb:70]: Thank you for being picky! Totally overlooked that.

**Comment by kjduensing** [comment]
@ryan-mcneil sorry about this - I updated a typo and it wiped your approval. Would you mind taking another quick look? Thanks!

**Pull request merged**